src/shared/port-provider.js

```diff
@@ -192,7 +164,13 @@ export class PortProvider {
           : new MessageChannel();
 
       const message = { frame1, frame2, type: 'offer' };
-      source.postMessage(message, origin, [messageChannel.port1]);
+      try {
+        source.postMessage(message, origin, [messageChannel.port1]);
+      } catch (err) {
+        throw new Error(
```

Comment:
I think this is a genuine problem that should crash the application. However, this is very unlikely that will err on a modern browser. Therefore, I would still suggest using `sendError` here.